In [1]:
import pandas as pd

In [2]:
titles = pd.read_csv('titles.csv')
dates = pd.read_csv("release_dates.csv")
casts = pd.read_csv('cast.csv')

### How many movies are listed in the titles dataframe?

In [3]:
titles.title.nunique()

223828

### What are the earliest two films listed in the titles dataframe?

In [4]:
titles.sort_values('year').head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

### How many movies have the title "Hamlet"?

In [5]:
titles[titles['title']=='Hamlet'].shape[0]

20

### How many movies are titled "North by Northwest"?

In [6]:
titles[titles['title']=='North by Northwest'].shape[0]

1

### When was the first movie titled "Hamlet" made?

In [7]:
titles[titles['title']=='Hamlet'].sort_values('year').head(1).year

52521    1910
Name: year, dtype: int64

### List all of the "Treasure Island" movies from earliest to most recent.

In [8]:
titles[titles['title']=='Treasure Island'].sort_values('year')

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [9]:
titles[titles.year == 1950].shape[0]

1113

Total 1113 movies were released in the year 1950

### How many movies were made in the year 1960?

In [10]:
titles[titles.year == 1960].shape[0]

1549

Total 1113 movies were released in the year 1960

### How many movies were made from 1950 through 1959?

In [11]:
titles[(titles.year>=1950) & (titles.year<1959)].shape[0]

11419

### In what years has a movie titled "Batman" been released?

In [12]:
titles[titles.title == 'Batman'].year

39349     1943
240896    1989
Name: year, dtype: int64

### How many roles were there in the movie "Inception"?

In [13]:
casts[casts['title']=='Inception'].character.shape[0]

78

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [14]:
casts[casts['title']=='Inception'].isnull().sum().sum()

27

### But how many roles in the movie "Inception" did receive an "n" value?

In [15]:
casts[casts['title']=='Inception'].shape[0] - casts[casts['title']=='Inception'].isnull().sum().sum()

51

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [16]:
temp = casts[(casts['title']=='North by Northwest') & (casts['n'].notnull())]

In [17]:
temp[['name','n']].sort_values('n')

name     n
871426             Cary Grant   1.0
3518333       Eva Marie Saint   2.0
1457165           James Mason   3.0
3157232   Jessie Royce Landis   4.0
356707         Leo G. Carroll   5.0
3048272  Josephine Hutchinson   6.0
1695173           Philip Ober   7.0
1275337         Martin Landau   8.0
2447081         Adam Williams   9.0
1811387          Edward Platt  10.0
665768      Robert Ellenstein  11.0
2295714          Les Tremayne  12.0
464736        Philip Coolidge  13.0
1508344         Patrick McVey  14.0
206471           Edward Binns  15.0
1384780             Ken Lynch  16.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [18]:
temp = casts[(casts['title']=='Sleuth') & (casts['n'].notnull()) & (casts['year']==1972)]

In [19]:
temp[['name','n']].sort_values('n')

name    n
1705394    Laurence Olivier  1.0
326773        Michael Caine  2.0
373876       Alec Cawthorne  3.0
1465483  John (II) Matthews  4.0
2724553  Eve (III) Channing  5.0
1448690        Teddy Martin  6.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [20]:
temp = casts[(casts['title']=='Sleuth') & (casts['n'].notnull()) & (casts['year']==2007)]

In [21]:
temp[['name','n']].sort_values('n')

name    n
326774   Michael Caine  1.0
1293634       Jude Law  2.0
1805907  Harold Pinter  3.0

### How many roles were credited in the silent 1921 version of Hamlet?

In [22]:
casts[(casts['title']=='Hamlet') & (casts['year']==1921)].character.shape[0]

9

### How many roles were credited in Branagh’s 1996 Hamlet?

In [23]:
casts[(casts['title']=='Hamlet') & (casts['year']==1996)].character.shape[0]

54

### How many "Hamlet" roles have been listed in all film credits through history?

In [24]:
casts[casts['title']=='Hamlet'].character.shape[0]

342

### How many people have played an "Ophelia"?

In [25]:
casts[casts['title']=='Ophelia'].character.shape[0]

20

### How many people have played a role called "The Dude"?

In [26]:
casts[casts['character']=='The Dude'].character.shape[0]

19

### How many people have played a role called "The Stranger"?

In [27]:
casts[casts['character']=='The Stranger'].character.shape[0]

212

### How many roles has Sidney Poitier played throughout his career?

In [28]:
casts[casts['name']=='Sidney Poitier'].shape[0]

43

### How many roles has Judi Dench played?

In [29]:
casts[casts['name']=='Judi Dench'].shape[0]

55

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [30]:
casts[(casts['name']=='Cary Grant') & (casts['year']==1940) & (casts['n']==2)]

title  year        name   type   character    n
871423  My Favorite Wife  1940  Cary Grant  actor  Nick Arden  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [31]:
casts[(casts['name']=='Cary Grant') & (casts['year']==1940) & (casts['n']==1)].character

871407          Walter Burns
871448           Matt Howard
871450    C. K. Dexter Haven
Name: character, dtype: object

### How many roles were available for actors in the 1950s?

In [32]:
casts[(casts['year']==1950) & (casts['type']=='actor')].character.shape[0]

16760

### How many roles were available for actresses in the 1950s?

In [33]:
casts[(casts['year']==1950) & (casts['type']=='actress')].character.shape[0]

5637

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [34]:
casts[(casts['year']<1980) & (casts['n']==1)].shape[0]

63957

### How many non-leading roles were available through from the beginning of film history through 1980?

In [35]:
casts[(casts['year']<1980) & (casts['n']!=1)].shape[0]

1095378

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [36]:
casts[(casts['year']<=1980) & (casts['n'].isnull())].shape[0]

448347